test/core/xds/xds_client_test.cc

```diff
@@ -6021,6 +6021,127 @@ TEST_F(XdsClientTest, FallbackOnStartup) {
                /*resource_names=*/{"foo1"});
 }
 
+// flow steps :
+// 1. we  get valid resource
+// 2. we recieved invalid resource
+// 3. we receive valid resource again
+// Test that we redeliver resources to watcher
+// so they know error is cleared
+TEST_F(XdsClientTest, RedeliverSameResourceAfterError) {
+  InitXdsClient();
+
```

Comment:
Nit, for future reference: Please remove blank lines within functions.  In general, my feeling is that if you really need to break up sections of the code into logical chunks, you're better off adding a one-line comment indicating what the next chunk is doing rather than just having a blank line, since that helps readers understand the intent of the code.  Conversely, if something is really so obvious that it doesn't need the one-line comment, then it probably doesn't need to be broken up into a separate section anyway.